### PR TITLE
feature (handlers/authmethods): add handlers for ldap auth method operations

### PR DESCRIPTION
**Pull request by jimlambrt**

### CODE REVIEW

File: internal/daemon/controller/handlers/authmethods/ldap.go [hunk 1]
```diff
@@ -0,0 +1,364 @@
...
+	if err != nil {
+		return nil, errors.Wrap(ctx, err, op)
+	}
+	atFn := func() (ldap.AuthTokenCreator, error) {
+		return atRepo, nil
+	}
+
+	rawTk, err := ldap.Authenticate(ctx, ldapFn, iamFn, atFn, req.GetAuthMethodId(), reqAttrs.GetLoginName(), reqAttrs.GetPassword())
+	if err != nil {
+		// let's not send back too much info about the error
```
Comment (johanbrandhorst) [inline internal/daemon/controller/handlers/authmethods/ldap.go:86]: Is the full error captured in the audit logs or does this truncate that error too?

File: internal/db/schema/migrations/oss/postgres/1000/01_ldap.up.sql [hunk 2]
```diff
@@ -232,6 +232,28 @@ comment on table auth_ldap_group_entry_search is
...
+create function auth_ldap_method_group_search() returns trigger 
+as $$
+declare
+  n integer;
+begin
+  if new.enable_groups = true and  new.use_token_groups = false then 
+    select into n count(*) from auth_ldap_group_entry_search where ldap_method_id = new.public_id;
+     if n < 1 then 
+        raise exception 'During % of auth_ldap_method public_id=% must have a configure group_dn when enable_groups = true and use_token_groups = false',tg_op,new.public_id;
+      end if;
```
Comment (johanbrandhorst) [inline internal/db/schema/migrations/oss/postgres/1000/01_ldap.up.sql:244]: The whitespace for these 3 lines should probably be consistent?

File: internal/auth/ldap/repository_auth_method_update.go [hunk 3]
```diff
@@ -43,6 +44,24 @@ const (
 	GroupNamesField           = "GroupNames"
 )
 
+// isEmpty returns true if all the args are empty.  Only supports checking
+// strings and pointers, all other types are assumed to be empty.
+func isEmpty(args ...any) bool {
```
[the remark this answers is not in the document]
Comment (jimlambrt) [reply internal/auth/ldap/repository_auth_method_update.go:49]: Sort of thinking `!isEmpty(...)` is far more readable than an inversion of it (like the bits you proposed).

File: internal/daemon/controller/handlers/authmethods/ldap.go (hunk 1 above)
Comment (jimlambrt) [reply internal/daemon/controller/handlers/authmethods/ldap.go:86]: audit events were/are generated when the error is originally created so it's fully captured.

File: internal/db/schema/migrations/oss/postgres/1000/01_ldap.up.sql (hunk 2 above)
Comment (jimlambrt) [reply internal/db/schema/migrations/oss/postgres/1000/01_ldap.up.sql]: done.

File: internal/auth/ldap/repository_auth_method_update.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (johanbrandhorst) [reply internal/auth/ldap/repository_auth_method_update.go:49]: Fair enough, I found it a bit hard to parse but lets leave it.
